lib/support/clone_vm.rb

```diff
@@ -428,12 +428,83 @@ def find_datacenter
       root_folder.childEntity.grep(RbVmomi::VIM::Datacenter).find { |x| x.name == datacenter }
     end
 
+    def customization_spec
+      unless options[:guest_customization]
+        return false
+      end
+
+      block = /\d{,2}|1\d{2}|2[0-4]\d|25[0-5]/
```

Comment:
I would probably prefer using IPAddr from the standard library instead of regex. That should make it a bit more understandable. Something like this maybe?

```
def ip?(string)
  IPAddr.new(string)
  true
rescue IPAddr::InvalidAddressError
  false
end
```

Then it'd be more like `unless ip?(options[:guest_customization][:ipAddress])`

Reply:
There's a regex that matches an ipv4 address in the stdlib as well :https://github.com/ruby/ruby/blob/1fb4e28002327c1224c3ed32783160b011f14747/lib/resolv.rb#L2354-L2358

Reply:
Thank you @tecracer-theinen as that works exactly as expected. I have not done a lot in Ruby and was uncertain of what was possible without needing to require additional. First real contribution for me.